File: api/src/main/java/com/mystore/manager/api/business/payload/StockMovementPayload.java
package com.mystore.manager.api.business.payload;

import com.mystore.manager.api.common.payload.GlobalUserDatePayload;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class StockMovementPayload extends GlobalUserDatePayload {
    private Integer productId;
    private String productCode;
    private String productName;
    private String movementType;
    private Double quantity;
    private String reason;
    private String movementDate;
}

File: ui-admin/src/app/backend/payloads/business/stock-movement-payload.ts
export class StockMovementPayload {
    id: number;
    productId: number;
    productCode: string;
    productName: string;
    categoryId: number;
    categoryName: string;
    posId: number;
    posCode: string;
    posName: string;
    movementType: string;
    quantity: number;
    reason: string;
    movementDate: string;
    createdAt: string;
    updatedAt: string;
    createdByFullName: string;
    updatedByFullName: string;

    constructor() {
        this.id = null;
        this.productId = null;
        this.productCode = null;
        this.productName = null;
        this.categoryId = null;
        this.categoryName = null;
        this.posId = null;
        this.posCode = null;
        this.posName = null;
        this.movementType = null;
        this.quantity = null;
        this.reason = null;
        this.movementDate = null;
        this.createdAt = null;
        this.updatedAt = null;
        this.createdByFullName = null;
        this.updatedByFullName = null;
    }
}
